.github/workflows/buildAndDeployPhar.yml

```diff
@@ -0,0 +1,44 @@
+# https://help.github.com/en/categories/automating-your-workflow-with-github-actions
+
+name: Build and deploy PHAR
+on:
+  push:
+    tags:
+    - "*"
+jobs:
+  build:
+    name: Build PHAR-File
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
+
+    - name: Build Phar
+      uses: MilesChou/composer-action/7.2/install@master
+      with:
+        args: buildphar
+
+    - name: Sign Phar
+      env:
+        DECRYPT_KEY: ${{ secrets.DECRYPT_KEY }}
+        SIGN_KEY: ${{ secrets.SIGN_KEY }}
+      run: |
+        export GPG_TTY=$(tty)
+        echo $DECRYPT_KEY
+        echo Decrypting key
+        gpg --batch --yes --passphrase $DECRYPT_KEY KeepAChangelogSigning.asc.gpg && gpg --batch --yes --import KeepAChangelogSigning.asc
+        echo Signing Artifact
+        gpg --batch --yes --pinentry-mode loopback --passphrase $SIGN_KEY -u 5A30219F03D3A2A5 --armor --detach-sig build/keep-a-changelog.phar
+
+    - name: Create release
```

Comment:
This part might be more problematic... as I generally create the release with keep-a-changelog itself!

How would I ensure we get the changelog for the release used as the release message as well?

Reply:
So currently you are doing the release yourself manually after you create and push the tag? 

Perhaps that can happen automatically via the github action. So all you have to do is create and push the tag and the github action then takes the relevant section from the Changelog.md and uses that as Release-Comment.

That does not solve the problem of having a release-name other than the tag though. Unless we can find a convention. But from what I've seen so far it's always "keep-a-changelog $tag"

Reply:
Have a look at https://github.com/heiglandreas/keep-a-changelog/releases/tag/2.2.1 . That release was created only by tagging the release localy and then pushing it to github. The actual release was then done completely by itself.

Reply:
That would be perfect!

Yeah, what I was doing before was:

```bash
$ keep-a-changelog version:tag {version}
$ keep-a-changelog version:release {version}
```

With the workflow you're proposing, this will become:

```bash
$ keep-a-changelog version:tag {version}
$ git push origin {version}
```

Correct?

Reply:
Correct!

Reply:
Or `git push --tags origin`